lib/timers.js

```
@@ -418,6 +404,27 @@ function processImmediate() {
 }
 
 
+function tryOnImmediate(immediate, queue) {
+  var threw = true;
+  try {
+    immediate._onImmediate();
+    threw = false;
+  } finally {
+    if (threw === true) {
```

I would assume so. Only `try {} catch(e) {}` or `try {} finally { ... return; ... }` can swallow exceptions.